src/components/PostDetails.tsx

```diff
@@ -66,29 +35,22 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         postId: post.id,
       });
 
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(commentsActions.addComment(newComment));
     } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      dispatch(commentsActions.setError(true));
```

Comment:
BTW, you could create async Thunk for it as well and handle `fulfilled` and `rejected` states in extra reducers